File: game-server/src/main/java/work/fking/pangya/game/packet/handler/UpdateChatMacrosPacketHandler.java
package work.fking.pangya.game.packet.handler;

import io.netty.buffer.ByteBuf;
import work.fking.pangya.game.net.ClientGamePacketHandler;
import work.fking.pangya.game.GameServer;
import work.fking.pangya.game.player.Player;
import work.fking.pangya.networking.protocol.ProtocolUtils;

public class UpdateChatMacrosPacketHandler implements ClientGamePacketHandler {

    private static final int MAX_MACROS = 9;
    private static final int MACRO_LENGTH = 64;

    @Override
    public void handle(GameServer server, Player player, ByteBuf packet) {
        var macros = new String[MAX_MACROS];

        for (int i = 0; i < MAX_MACROS; i++) {
            macros[i] = ProtocolUtils.readFixedSizeString(packet, MACRO_LENGTH);
        }
        System.out.println(macros);
    }
}

File: game-server/src/main/kotlin/work/fking/pangya/game/packet/handler/UpdateChatMacrosPacketHandler.kt
package work.fking.pangya.game.packet.handler

import io.netty.buffer.ByteBuf
import work.fking.pangya.game.GameServer
import work.fking.pangya.game.net.ClientPacketHandler
import work.fking.pangya.game.player.Player
import work.fking.pangya.networking.protocol.readFixedSizeString

class UpdateChatMacrosPacketHandler : ClientPacketHandler {
    companion object {
        private const val MAX_MACROS = 9
        private const val MACRO_LENGTH = 64
    }

    override fun handle(server: GameServer, player: Player, packet: ByteBuf) {
        val macros = arrayOfNulls<String>(MAX_MACROS)
        for (i in 0 until MAX_MACROS) {
            macros[i] = packet.readFixedSizeString(MACRO_LENGTH)
        }
    }
}
